### PR TITLE
improve hydra.utils.get_{class,method,object}

### LINKED ISSUE #1972: Document hydra.utils.{get_class,get_method}
Users would benefit from documentation for the following:
[`hydra.utils.get_class`](https://github.com/facebookresearch/hydra/blob/1cbe86ebecbeb134a3f2041120d57447a7394314/hydra/utils.py#L21)
[`hydra.utils.get_method`](https://github.com/facebookresearch/hydra/blob/1cbe86ebecbeb134a3f2041120d57447a7394314/hydra/utils.py#L32)

### LINKED ISSUE #2139: [Feature Request] `hydra.utils.get_object` method
# 🚀 Feature Request

The `hydra.utils` module exposes `get_class` and `get_method` functions that can be used to look up callable objects:
```python
>>> from hydra.utils import get_method
>>> get_method("torch.tensor")
<built-in method tensor of type object at 0x7fdbd8ff1760>
```
To resolve the given string, these methods use the same backend as `hydra.utils.instantiate`.

The `get_class` and `get_method` functions error if the given dotpath does not point to a callable:
```python
>>> get_method("torch.double")
...
ValueError: Located non-callable of type 'dtype' while loading 'torch.double'
```

I think it would be useful to expose a `get_object` function in `hydra.utils` that does *not* give an error in this case. This would allow users to leverage Hydra's dotpath-lookup mechanism to discover arbitrary objects (rather than just callable ones).

### PR DESCRIPTION
**Pull request by Jasha10**
Implement `hydra.utils.get_object` for an easier way to look up objects from a dotpath.

Commits:
- improve err msg for get_class failure
- improve tests for get_{class,method}
- implement hydra.utils.get_object (Closes #2139)
- document hydra.utils.get_{class,method,object} (Closes #1972)

### CODE REVIEW

**Comment by omry** [comment]
Duh. didn't submit my pending comment.

File: hydra/utils.py [hunk 1]
```diff
@@ -51,6 +67,22 @@ def get_method(path: str) -> Callable[..., Any]:
...
+    """
+    Look up a callable based on a dotpath.
+
+    >>> import my_module
+    >>> from hydra.utils import get_object
+    >>> assert get_object("my_module.my_object") is my_module.my_object
+    """
+    try:
+        obj = _locate(path)
+        return obj
```
[the remark this answers is not in the document]
Comment (Jasha10) [reply hydra/utils.py:80]: Yes, exactly!

File: hydra/utils.py [hunk 2]
```diff
@@ -51,6 +67,22 @@ def get_method(path: str) -> Callable[..., Any]:
 get_static_method = get_method
 
 
+def get_object(path: str) -> Any:
+    """
+    Look up a callable based on a dotpath.
```
Comment (omry) [inline hydra/utils.py:72]: the comment he is incorrect. object may not be a callable.

Comment (Jasha10) [reply hydra/utils.py:72]: I've opened PR #2544 with a fixup.

**Comment by Jasha10** [comment]
Closes #1975

**Pull request merged**